src/dawn/engine_dawn.cc

```diff
@@ -27,10 +28,14 @@ Result EngineDawn::Initialize() {
   if (device_)
     return Result("Dawn:Initialize device_ already exists");
 
-  // The constructor already gives us a default device.  So there is nothing
-  // left to do.
+#if AMBER_DAWN_METAL
+  Result result = CreateMetalDevice(&device_);
+  if (device_)
```

Comment:
You should just be able to return result without doing the check. That way the device creation error will propagate instead of getting lost.

(which means, this can just be `return CreateMetalDevice(&device_);`

Reply:
Right.  This is a vestige of my debugging.